docs/protocol/network/messages/inventory.md

```diff
@@ -0,0 +1,163 @@
+           |                                |
+           |            GET_DATA            |
+           +------------------------------->+
+           |             BLOCK              |
+           +<-------------------------------+
+           |             BLOCK              |
+           +<-------------------------------+
+           |             BLOCK              |
+           +<-------------------------------+
+           |                                |
+           +                                +
+```
+
+## Inventory Broadcasting
+
+Similarly to the previously described process of synchronization, any node may contribute to the synchronization of their outbound peers by advertising inventory objects such as blocks and transactions. Inventory broadcasting is also used in case a node creates transactions or mine blocks.
+
+The inventory broadcasting can be described as the following sequence of steps:
+
+ 1. A remote node broadcasts its inventory by sending an `inv` message, containing all the hashes of the advertised inventory objects.
+
+ 2. After receiving an `inv` message and filtering the inventory objects that may be missing in the local blockchain, the local node sends a `get_data` message with the hashes of the needed objects.
+
+ 3. The remote note receives the `get_data` message and sends a `block` or `tx` message per requested inventory object (identified by a hash).
+
+The following diagram depicts the previous step under the assumption that the local node (`NodeA`) sends a `get_data` message requesting 3 blocks and 2 transactions.
+
+```ascii
+           NodeA                            NodeB
+             +                                +
+             |              INV               |
+             +<-------------------------------+
+             |                                |
+             |           GET_DATA             |
+             +------------------------------->+
+             |                                |
+             |             BLOCK              |
+             +<-------------------------------+
+             |             BLOCK              |
+             +<-------------------------------+
+             |             BLOCK              |
+             +<-------------------------------+
+             |              TX                |
+             +<-------------------------------+
+             |              TX                |
+             +<-------------------------------+
+             |                                |
+             +                                +
+```
+
+## Get blocks message
+
+The `get_blocks` messages is used in order to notify the hash of the heighest known block by the peer. After the exchanging `get_blocks` messages between peers, the one with the longest blockchain in terms of blocks will send an `inv` message to the other peer. This message will include the list of blocks hashes starting right after the last known block hash provided by the other peer.
```

Comment:
The `get_blocks` messages are used in order to notify the hash of the highest known block by the peer. After exchanging `get_blocks` messages between peers, the one with the longest blockchain in terms of blocks will send an `inv` message to the other peer. This message will include the list of block hashes starting right after the last known block hash provided by the other peer.